WeatherApp-UIKit/WeatherApp-UIKit/WeatherViewModelTests.swift

```diff
@@ -14,151 +14,270 @@ import XCTest
-
+    
     XCTAssertEqual(result.events, [
       .init(time: 0, value: .next(.loading)),
       .init(time: 100, value: .next(.error)),
     ])
   }
-
+  
   func testLoadingIsPropagated() {
     let location = locationObservable(with: [
       .init(time: 0, value: .next(.location(mockLocations[0]))),
     ])
-
+    
     let api: WeatherViewModel.API = { _ in
       self.scheduler
         .createColdObservable([.next(500, Weather.mock), .completed(500)])
         .asObservable()
     }
-
+    
     let sut = WeatherViewModel(weatherAPI: api, locale: locale, currentLocation: location)
     let result = runAndObserve(sut.state)
-
+    
     XCTAssertEqual(result.events,
                    [
-                     .init(time: 0, value: .next(.loading)),
-                     .init(time: 500, value: .next(.loadedMock)),
-                   ])
+                    .init(time: 0, value: .next(.loading)),
+                    .init(time: 500, value: .next(.loadedMock)),
+    ])
   }
-
+  
   func testSuccess() {
     let location = locationObservable(with: [
       .init(time: 0, value: .next(.loading)),
       .init(time: 100, value: .next(.location(mockLocations[0]))),
     ])
-
+    
     let api: WeatherViewModel.API = { _ in
       self.scheduler
         .createColdObservable([.init(time: 100, value: .next(Weather.mock))])
         .asObservable()
     }
-
+    
     let sut = WeatherViewModel(weatherAPI: api, locale: locale, currentLocation: location)
     let result = runAndObserve(sut.state)
-
+    
     XCTAssertEqual(result.events, [
       .init(time: 0, value: .next(.loading)),
       .init(time: 100, value: .next(.loading)),
       .init(time: 200, value: .next(.loadedMock)),
     ])
   }
-
+  
   func testLikeDislikeButtonsPress() {
     let location = locationObservable(with: [
       .init(time: 0, value: .next(.location(mockLocations[0]))),
     ])
-
+    
     let api: WeatherViewModel.API = { _ in .just(.mock) }
-
+    
     var likedWeather: Weather?
     var dislikedWeather: Weather?
-
+    
     let storage: WeatherViewModel.Storage = { _weather, liked in
       if liked {
         likedWeather = _weather
       } else {
         dislikedWeather = _weather
       }
     }
-
+    
     let sut = WeatherViewModel(weatherAPI: api, storage: storage, locale: locale, currentLocation: location)
-
+    
     let likeButtonTap = scheduler
       .createColdObservable([.next(100, ())])
       .asDriver(onErrorJustReturn: ())
-
+    
     likeButtonTap
       .drive(sut.likeButtonTapped)
       .disposed(by: disposeBag)
-
+    
     let dislikeButtonTap = scheduler
       .createColdObservable([.next(200, ())])
       .asDriver(onErrorJustReturn: ())
-
+    
     dislikeButtonTap
       .drive(sut.dislikeButtonTapped)
       .disposed(by: disposeBag)
-
+    
     scheduler.start()
+    
+    XCTAssertEqual(likedWeather, Weather.mock)
+    XCTAssertEqual(dislikedWeather, Weather.mock)
+  }
+  
+  // MARK: new tests
```

Comment:
Why do you write new tests? I thought you would use the existing ones and just adjust them.